src/lib/pubkey/dilithium/dilithium_polynomials.h

```diff
@@ -0,0 +1,1566 @@
+      *
+      * Arguments:   - Polynomial& r: pointer to output polynomial
+      *              - const uint8_t *a: byte array with bit-packed_t1 polynomial
+      *           - const DilithiumModeConstants& mode: reference to dilihtium mode values
+      **************************************************/
+      static void polyz_unpack(Polynomial* r, const uint8_t* a, const DilithiumModeConstants& mode)
+      {
+         if (mode.get_gamma1() == (1 << 17))
+         {
+            for (unsigned int i = 0; i < DilithiumModeConstants::N / 4; ++i)
+            {
+               r->m_coeffs[4 * i + 0] = a[9 * i + 0];
+               r->m_coeffs[4 * i + 0] |= static_cast<uint32_t>(a[9 * i + 1]) << 8;
+               r->m_coeffs[4 * i + 0] |= static_cast<uint32_t>(a[9 * i + 2]) << 16;
+               r->m_coeffs[4 * i + 0] &= 0x3FFFF;
+
+               r->m_coeffs[4 * i + 1] = a[9 * i + 2] >> 2;
+               r->m_coeffs[4 * i + 1] |= static_cast<uint32_t>(a[9 * i + 3]) << 6;
+               r->m_coeffs[4 * i + 1] |= static_cast<uint32_t>(a[9 * i + 4]) << 14;
+               r->m_coeffs[4 * i + 1] &= 0x3FFFF;
+
+               r->m_coeffs[4 * i + 2] = a[9 * i + 4] >> 4;
+               r->m_coeffs[4 * i + 2] |= static_cast<uint32_t>(a[9 * i + 5]) << 4;
+               r->m_coeffs[4 * i + 2] |= static_cast<uint32_t>(a[9 * i + 6]) << 12;
+               r->m_coeffs[4 * i + 2] &= 0x3FFFF;
+
+               r->m_coeffs[4 * i + 3] = a[9 * i + 6] >> 6;
+               r->m_coeffs[4 * i + 3] |= static_cast<uint32_t>(a[9 * i + 7]) << 2;
+               r->m_coeffs[4 * i + 3] |= static_cast<uint32_t>(a[9 * i + 8]) << 10;
+               r->m_coeffs[4 * i + 3] &= 0x3FFFF;
+
+               r->m_coeffs[4 * i + 0] = mode.get_gamma1() - r->m_coeffs[4 * i + 0];
+               r->m_coeffs[4 * i + 1] = mode.get_gamma1() - r->m_coeffs[4 * i + 1];
+               r->m_coeffs[4 * i + 2] = mode.get_gamma1() - r->m_coeffs[4 * i + 2];
+               r->m_coeffs[4 * i + 3] = mode.get_gamma1() - r->m_coeffs[4 * i + 3];
+            }
+         }
+         else if (mode.get_gamma1() == (1 << 19))
+         {
+            for (unsigned int i = 0; i < DilithiumModeConstants::N / 2; ++i) {
+               r->m_coeffs[2 * i + 0] = a[5 * i + 0];
+               r->m_coeffs[2 * i + 0] |= static_cast<uint32_t>(a[5 * i + 1]) << 8;
+               r->m_coeffs[2 * i + 0] |= static_cast<uint32_t>(a[5 * i + 2]) << 16;
+               r->m_coeffs[2 * i + 0] &= 0xFFFFF;
+
+               r->m_coeffs[2 * i + 1] = a[5 * i + 2] >> 4;
+               r->m_coeffs[2 * i + 1] |= static_cast<uint32_t>(a[5 * i + 3]) << 4;
+               r->m_coeffs[2 * i + 1] |= static_cast<uint32_t>(a[5 * i + 4]) << 12;
+               r->m_coeffs[2 * i + 0] &= 0xFFFFF;
+
+               r->m_coeffs[2 * i + 0] = mode.get_gamma1() - r->m_coeffs[2 * i + 0];
+               r->m_coeffs[2 * i + 1] = mode.get_gamma1() - r->m_coeffs[2 * i + 1];
+            }
+         }
```

Comment:
Using the dimension makes sense. if possible, I would use `switch` without a default case. That should be fine for the compiler, as long as all enum states are covered. Otherwise default and `BOTAN_ASSERT()`, maybe?